### PR TITLE
CKS-531 Add Ip Allowlisting for georestriction

### PR DESCRIPTION
**Pull request by colinmazu**

### CODE REVIEW

File: terraform/deploy-static-to-s3.sh [hunk 1]
```diff
@@ -79,3 +79,22 @@ aws s3 cp $pathToStaticFiles s3://$s3BucketName/$releaseNumber \
   --exclude "*.woff2" \
   --exclude "/static/*" \
   --recursive --no-progress
+
+aws s3 cp $pathToStaticFiles s3://$s3BucketName/$releaseNumber \
```
Comment (ediblecode) [inline terraform/deploy-static-to-s3.sh:83]: Isn't this doing the same as the above?

File: terraform/modules/cf_hosting/cloudfront.tf [hunk 2]
```diff
@@ -18,7 +23,94 @@ resource "aws_cloudfront_distribution" "s3_distribution" {
...
+		lambda_function_association {
+			event_type = "viewer-request"
+			lambda_arn = var.viewer_request_edge_lambda_qualified_arn
+		}
+
+		compress               = true
+		viewer_protocol_policy = "redirect-to-https"
+	}
+
+	ordered_cache_behavior {
```
Comment (ediblecode) [inline terraform/modules/cf_hosting/cloudfront.tf:55]: I think these would be useful to have a comment explaining why we have a cache behaviour for each static file type - to basically explain that we only need to IP restrict HTML pages, which mean assets are faster to serve and cheaper

File: terraform/modules/cf_hosting/cloudfront.tf [hunk 3]
```diff
@@ -18,7 +23,94 @@ resource "aws_cloudfront_distribution" "s3_distribution" {
...
+		forwarded_values {
+			query_string = false
+			headers      = ["CloudFront-Viewer-Country"]
+
+			cookies {
+				forward = "none"
+			}
+		}
+		lambda_function_association {
+			event_type = "origin-request"
```
Comment (ediblecode) [inline terraform/modules/cf_hosting/cloudfront.tf:71]: Do we even need to run the origin request and response lambdas on static files that aren't HTML?

Comment (colinmazu) [reply terraform/modules/cf_hosting/cloudfront.tf:71]: yeah, we need the response headers I guess as they add the security info (do we?). But agree as long as we don't mind the static files not being restricted that its fine (and prob a better option as less cost)

File: edge-lambdas/viewer-request-edge-lambda/ipAllowList.json [hunk 4]
```diff
@@ -0,0 +1 @@
+["169.254.1.1/24", "169.254.1.2/24", "169.254.1.3/24"]
```
[the remark this answers is not in the document]
Comment (colinmazu) [reply edge-lambdas/viewer-request-edge-lambda/ipAllowList.json:1]: was just going to use octo string replace feature

File: terraform/deploy-static-to-s3.sh (hunk 1 above)
Comment (colinmazu) [reply terraform/deploy-static-to-s3.sh:83]: fair point

File: edge-lambdas/viewer-request-edge-lambda/ipAllowList.json (hunk 4 above)
[the remark this answers is not in the document]
Comment (colinmazu) [reply edge-lambdas/viewer-request-edge-lambda/ipAllowList.json:1]: add some logic to detect if function is running locally or not - and feed in octo variable into function

File: terraform/modules/cf_hosting/cloudfront.tf (hunk 2 above)
Comment (colinmazu) [reply terraform/modules/cf_hosting/cloudfront.tf:55]: explain why....

**Pull request merged**